algo-core/ortools_solver.py

```diff
@@ -730,38 +1367,88 @@ def generate_schedule_with_ortools():
+        if len(agents_onb) > 0:
+            o_path = "onboarding_message.txt"
+            o_file = open(o_path, "w")
+            o_file.write(
+                "**Support agent onboarding next week**"
+                "\n\nEach new onboarding agent has been paired with a senior "
+                "support agent for each of their shifts. The senior agent "
+                "will act as a mentor for the onboarding agents, showing "
+                "them the ropes during these onboarding shifts (see the "
+                "[onboarding document](https://github.com/balena-io/process/blob/master/process/support/onboarding_agents_to_support.md) "
+                "for background). Here are the mentor-novice pairings "
+                "for next week:"
+            )
+
         for d in range(num_days):
+            if len(agents_onb) > 0:
+                o_file.write(
+                    f"\n\n**Onboarding on {days[d].strftime('%Y-%m-%d')}**"
+                )
+
             day_dict = {}
             day_dict["start_date"] = days[d]
             day_dict["shifts"] = []
 
-            for t in range(num_tracks):
-                for h in df_agents.index:
-                    if solver.Value(v_tdh.loc[(t, d, h), "Duration"]) != 0:
+            if d == 0:
+                max_tracks = num_tracks + 1
+            else:
+                max_tracks = num_tracks
+
+            for t in range(max_tracks):
+                for h in agents_vet:
+                    if (
+                        solver.Value(v_tdh.loc[(t, d, h), "shift_duration"])
+                        != 0
+                    ):
                         day_dict["shifts"].append(
                             (
                                 h,
-                                solver.Value(v_tdh.loc[(t, d, h), "Start"]),
-                                solver.Value(v_tdh.loc[(t, d, h), "End"]),
+                                solver.Value(
+                                    v_tdh.loc[(t, d, h), "shift_start"]
+                                ),
+                                solver.Value(
+                                    v_tdh.loc[(t, d, h), "shift_end"]
+                                ),
                             )
                         )
 
+            for h in agents_onb:
+                if solver.Value(v_dh_on.loc[(d, h), "shift_duration"]) != 0:
+                    day_dict["shifts"].append(
+                        (
+                            h,
+                            solver.Value(v_dh_on.loc[(d, h), "shift_start"]),
+                            solver.Value(v_dh_on.loc[(d, h), "shift_end"]),
+                        )
+                    )
+
+                for m in v_mentors.columns:
+                    if solver.Value(v_mentors.loc[(d, h), m]) == 1:
+                        o_file.write(f"\n{m} will mentor {h}")
+
             schedule_results.append(day_dict)
+        if len(agents_onb) > 0:
+            o_file.write("\n\ncc @GeorgiaTs")
```

Comment:
I have the same comment here to remove such references to balena specific team members.